common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -276,10 +277,19 @@ public void initChannel(SocketChannel ch) {
     // Connect to the remote server
     long preConnect = System.nanoTime();
     ChannelFuture cf = bootstrap.connect(address);
-    if (!cf.await(conf.connectionCreationTimeoutMs())) {
+
+    if (connCreateTimeout <= 0) {
```

Comment:
Although negative cases are prevented already via `connectionCreationTimeoutMs` changes, this condition looks safer.